src/discussions/posts/post-editor/PostEditor.jsx

```diff
@@ -1,134 +1,304 @@
+        touched,
+        handleSubmit,
+        handleBlur,
+        handleChange,
+      }) => (
+        <Form className="mx-4 my-2" onSubmit={handleSubmit}>
+          <h3>
+            {editExisting
+              ? intl.formatMessage(messages.editPostHeading)
+              : intl.formatMessage(messages.addPostHeading)}
+          </h3>
+          <Form.RadioSet
+            name="postType"
+            className="d-flex flex-row my-3"
+            value={values.postType}
+            onChange={handleChange}
+            onBlur={handleBlur}
+            aria-label={intl.formatMessage(messages.postTitle)}
+          >
+            <DiscussionPostType
+              value="discussion"
+              selected={values.postType === 'discussion'}
+              type={intl.formatMessage(messages.discussionType)}
+              icon={<Post />}
+              description={intl.formatMessage(messages.discussionDescription)}
+            />
+            <DiscussionPostType
+              value="question"
+              selected={values.postType === 'question'}
+              type={intl.formatMessage(messages.questionType)}
+              icon={<Help />}
+              description={intl.formatMessage(messages.questionDescription)}
+            />
+          </Form.RadioSet>
+          <Form.Group className="py-2 w-50">
+            <Form.Control
+              name="topic"
+              as="select"
+              value={values.topic}
+              onChange={handleChange}
+              onBlur={handleBlur}
+              aria-describedby="topicAreaInput"
+              floatingLabel={intl.formatMessage(messages.topicArea)}
+            >
+              {nonCoursewareTopics.map(topic => (
+                <option key={topic.id} value={topic.id}>{topic.name}</option>
+              ))}
+              {coursewareTopics.map(topic => (
+                <optgroup label={topic.name} key={topic.name}>
+                  {topic.children.map(subtopic => (
+                    <option
+                      key={subtopic.id}
+                      value={subtopic.id}
+                    >{subtopic.name}
+                    </option>
+                  ))}
+                </optgroup>
+              ))}
+            </Form.Control>
+          </Form.Group>
+          <div className="border-bottom my-4" />
+          <Form.Group className="py-2" isInvalid={Boolean(touched.title && errors.title)}>
+            <Form.Control
+              name="title"
+              type="text"
+              onChange={handleChange}
+              onBlur={handleBlur}
+              aria-describedby="titleInput"
+              floatingLabel={intl.formatMessage(messages.postTitle)}
+              value={values.title}
+            />
+            {(touched.title && errors.title)
+            && (
+              <Form.Control.Feedback type="invalid" hasIcon={false}>
+                {intl.formatMessage(messages.titleError)}
+              </Form.Control.Feedback>
+            )}
```

Comment:
We can make it much cleaner as validation errors display code is repeating. So we can create a function.

const isValidTitle = Boolean(touched.title && errors.title)
{isValidTitle && ( displayError({intl.formatMessage(messages.commentError)}) )}
            
 const displayError = (message) => (
      <Form.Control.Feedback type="invalid" hasIcon={false}>
                {message}
       </Form.Control.Feedback>
)